### PR TITLE
Prettify concat and downstream expressions.

### PR DESCRIPTION
**Pull request by jeschkies**
**What this PR does / why we need it**:
Prettifying `ConcatSampleExpr` et al will simplify debugging the shard mapper.

**Checklist**
- [ ] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [x] Tests updated
- [ ] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

**Review by kavirajk — COMMENTED** [review]
Looks good overall 👍. Left one minor comment. 

P.S: would have been nicer, if these types were also in `syntax` package so that all the `Pretty` implementation would be in single file `prettier.go`. But not a big deal, also given these types are part of downstream engine and not part of "user" facing LogQL, make sense to add it in the `downstream.go`

Thanks @jeschkies :)

File: pkg/logql/downstream.go [hunk 1]
```diff
@@ -83,6 +83,22 @@ func (d DownstreamSampleExpr) String() string {
 	return fmt.Sprintf("downstream<%s, shard=%s>", d.SampleExpr.String(), d.shard)
 }
 
+func (d DownstreamSampleExpr) Pretty(level int) string {
```
[the remark this answers is not in the document]
Comment (jeschkies) [reply pkg/logql/downstream.go:86]: They do not have a logql representation. I'll add this in the comments.

**Comment by jeschkies** [comment]
> P.S: would have been nicer, if these types were also in syntax package so that all the Pretty implementation would be in single file prettier.go.

@kavirajk I agree. This would be possible with an extended version of the [visitor](https://github.com/grafana/loki/blob/main/pkg/logql/syntax/visit.go). The `prettier.go` would have to move under `logql` though.

**Review by kavirajk — APPROVED** [review]
thanks @jeschkies. LGTM 👍

**Pull request merged**